### PR TITLE
<HELIOS> Folder Struct Changes

### PR DESCRIPTION
**Pull request by pranjalchanda08**

### CODE REVIEW

File: src/include/visor/terravisor/helios/helios_tasks.h [hunk 1]
```diff
@@ -158,7 +158,7 @@ void helios_task_yield(void);
  *
  * @return status_t
  */
-void helios_run(void);
+void helios_run(void) _NORETURN;
```
Comment (akashkollipara) [inline src/include/visor/terravisor/helios/helios_tasks.h:161]: i dont think you need to define it here.

**Review by akashkollipara — CHANGES_REQUESTED** [review]

File: src/include/visor/terravisor/helios/helios_tasks.h (hunk 1 above)
Comment (akashkollipara) [reply src/include/visor/terravisor/helios/helios_tasks.h:161]: this attribute should be used in C files.

File: src/visor/terravisor/services/kernel/helios/helios_mutex.c [hunk 2]
```diff
@@ -81,8 +81,8 @@ status_t helios_mutex_lock 	(helios_mutex_t * mutex_ptr, size_t wait_ticks)
 				g_sched_ctrl.curr_task->wait_res.wait_on_resource = (uintptr_t) mutex_ptr;
 				helios_task_wait(wait_ticks);
 			}
-			else if (mutex_ptr->lock_task == g_sched_ctrl.curr_task) {
-				/* If locking task is current task, allow locking recursively */
+			else {
+				/* If locking task is current task or HELIOS_NULL_PTR, allow locking recursively */
 				lock_flag = true;
 			}
 		}
```
Comment (akashkollipara) [inline src/visor/terravisor/services/kernel/helios/helios_mutex.c:88]: in consistency in `if-else {}`
either use
```C
if () {
} else {
}
```
or
```C
if ()
{
}
else
{
}
```

**Review by akashkollipara — CHANGES_REQUESTED** [review]

File: src/visor/terravisor/services/kernel/helios/helios_mutex.c [hunk 3]
```diff
@@ -112,6 +112,12 @@ status_t helios_mutex_unlock (helios_mutex_t * mutex_ptr)
 	else {
 		/* Unlock mutex only if current task requests it. */
 		mutex_ptr->mutex_val++;
+		if (mutex_ptr->mutex_val == MUTEX_INIT_VAL)
+		{
+			/* When Unlocked set current task to NULL */
+			mutex_ptr->lock_task = HELIOS_NULL_PTR;
+		}
+
```
Comment (Dybios) [inline src/visor/terravisor/services/kernel/helios/helios_mutex.c:120]: Thanks for handling this.

**Review by Dybios — APPROVED** [review]

File: src/visor/terravisor/services/kernel/helios/helios_mutex.c [hunk 4]
```diff
@@ -112,6 +114,12 @@ status_t helios_mutex_unlock (helios_mutex_t * mutex_ptr)
 	else {
```
Comment (akashkollipara) [inline src/visor/terravisor/services/kernel/helios/helios_mutex.c:114]: this line too...

**Review by akashkollipara — CHANGES_REQUESTED** [review]

**Review by akashkollipara — APPROVED** [review]

**Pull request merged**